src/streaming/XHRLoader.js

```diff
@@ -40,267 +38,52 @@ import ErrorHandler from './utils/ErrorHandler';
-                    }
-
-                    if (config.complete) {
-                        config.complete(request, xhr.statusText);
-                    }
-                }
-            }
-        };
-
-        const progress = function (event) {
-            let currentTime = new Date();
-
-            if (firstProgress) {
-                firstProgress = false;
-                if (!event.lengthComputable ||
-                    (event.lengthComputable && event.total !== event.loaded)) {
-                    request.firstByteDate = currentTime;
-                }
-            }
-
-            if (event.lengthComputable) {
-                request.bytesLoaded = event.loaded;
-                request.bytesTotal = event.total;
-            }
-
-            traces.push({
-                s: lastTraceTime,
-                d: currentTime.getTime() - lastTraceTime.getTime(),
-                b: [event.loaded ? event.loaded - lastTraceReceivedCount : 0]
-            });
-
-            lastTraceTime = currentTime;
-            lastTraceReceivedCount = event.loaded;
-
-            if (config.progress) {
-                config.progress();
-            }
-        };
+        let lastTraceTime = requestStartTime; /*jshint ignore:line*/
+        let lastTraceReceivedCount = 0; /*jshint ignore:line*/
 
-        const onload = function () {
-            if (xhr.status >= 200 && xhr.status <= 299) {
-                handleLoaded(true);
+        let request = httpRequest.request;
 
-                if (config.success) {
-                    config.success(xhr.response, xhr.statusText, xhr);
-                }
-
-                if (config.complete) {
-                    config.complete(request, xhr.statusText);
-                }
-            }
-        };
-
-        const onabort = function () {
-            if (config.abort) {
-                config.abort(request, xhr.status);
-            }
-        };
+        let xhr = new XMLHttpRequest();
+        xhr.open(httpRequest.method, httpRequest.url, true);
 
-        if (!requestModifier || !metricsModel || !errHandler) {
-            throw new Error('config object is not correct or missing');
+        if (request.responseType) {
+            xhr.responseType = request.responseType;
         }
 
-        try {
-            const modifiedUrl = requestModifier.modifyRequestURL(request.url);
-            const verb = request.checkExistenceOnly ? HTTPRequest.HEAD : HTTPRequest.GET;
-
-            xhr.open(verb, modifiedUrl, true);
-
-            if (request.responseType) {
-                xhr.responseType = request.responseType;
-            }
-
-            if (request.range) {
-                xhr.setRequestHeader('Range', 'bytes=' + request.range);
-            }
-
-            if (!request.requestStartDate) {
-                request.requestStartDate = requestStartTime;
-            }
-
-            xhr = requestModifier.modifyRequestHeader(xhr);
```

Comment:
This functionality seems to have been removed completely? This would break anyone relying on header authentication etc.

Reply:
You are right. This was removed when the loader abstraction layer was added. Needs to be restored.